src/app/features/auth/pages/login/login-page.component.ts

```diff
@@ -31,20 +31,22 @@ export class LoginPageComponent {
     private readonly _toastr: ToastrService
   ) {}
 
-  public async login(formData: LoginData): Promise<void> {
-    const [error] = await to(
-      firstValueFrom(
-        this._store.dispatch(new Login(formData, formData.remember))
+  public async login(formData: LoginCredentials): Promise<void> {
+    const [error] = await to<Login, HttpErrorResponse>(
+      firstValueFrom<Login>(
+        this._store
+          .dispatch(new Login(formData))
+          .pipe(catchError((e: HttpErrorResponse) => Promise.reject(e)))
```

Comment:
That is an odd way to handle an error. What exactly you are doing here? What changes if you remove this line?

Reply:
If I remove this line I will get a eslint error: ESLint: Unsafe argument of type `Observable<any>` assigned to a parameter of type `Observable<Login>`
So here I see two cases: 
1. As was done here - with generics

const [error] = await to<Login, HttpErrorResponse>(
      firstValueFrom<Login>(
        this._store
          .dispatch(new Login(formData))
          .pipe(catchError((e: HttpErrorResponse) => Promise.reject(e)))
      )
    )

    if (error) {
      const loginError = error.error as ErrorResponse
      this._toastr.error(loginError.message, loginError.error)
      return
    }

3. As it was implemented the previous time - using casting with as operator.
const [error] = await to(
      firstValueFrom(
        this._store.dispatch(new Login(formData, formData.remember))
      )
    )

    if (error) {
      const loginError = (error as HttpErrorResponse).error as ResponseError
      this._toastr.error(loginError.message, loginError.error )
      return
    }